pkg/provider/provider.go

```diff
@@ -38,16 +39,21 @@ type Provider struct {
 	identity *FullIdentity
 }
 
-// NewProvider creates a Provider out of an Identity, a net.Listener, and a set
-// of responsibilities.
-func NewProvider(identity *FullIdentity, lis net.Listener,
+// NewProvider creates a Provider out of an Identity, a net.Listener, a UnaryInterceptorProvider and
+// a set of responsibilities.
+func NewProvider(identity *FullIdentity, lis net.Listener, interceptor grpc.UnaryServerInterceptor,
 	responsibilities ...Responsibility) (*Provider, error) {
 	// NB: talk to anyone with an identity
 	ident, err := identity.ServerOption()
 	if err != nil {
 		return nil, err
 	}
 
+	unaryInterceptor := unaryInterceptor
+	if interceptor != nil {
+		unaryInterceptor = grpc_middleware.ChainUnaryServer(unaryInterceptor, interceptor)
```

Comment:
Don't pull in go-grpc-middleware, it has a quite a few dependencies (see https://gist.github.com/egonelbre/ef129c04b595641e61cfa3bcac1a2e4f). 

For combining you can use something like:
```
func CombineInterceptors(a, b grpc.UnaryServerInterceptor) grpc.UnaryServerInterceptor {
	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
		return a(ctx, req, info, func(actx context.Context, areq interface{}) (interface{}, error) {
			return b(actx, areq, info, func(bctx context.Context, breq interface{}) (interface{}, error) {
				return handler(bctx, breq)
			})
		})
	}
)
```

Reply:
Sure. Thanks for the snippet!